src/main/kotlin/io/sip3/captain/ce/pipeline/IcmpHandler.kt

```diff
@@ -17,48 +17,27 @@
 package io.sip3.captain.ce.pipeline
 
 import io.netty.buffer.ByteBuf
-import io.sip3.captain.ce.Routes
-import io.sip3.captain.ce.USE_LOCAL_CODEC
-import io.sip3.captain.ce.domain.ByteArrayPayload
 import io.sip3.captain.ce.domain.Packet
 import io.vertx.core.Vertx
 
 /**
  * Handles ICMP packets
  */
-class IcmpHandler(vertx: Vertx, bulkOperationsEnabled: Boolean) : Handler(vertx, bulkOperationsEnabled) {
-
-    private val packets = mutableListOf<Packet>()
-    private var bulkSize = 1
-
-    init {
-        if (bulkOperationsEnabled) {
-            vertx.orCreateContext.config().getJsonObject("icmp")?.let { config ->
-                config.getInteger("bulk-size")?.let { bulkSize = it }
-            }
-        }
-    }
+class IcmpHandler(val ipv4Handler: Ipv4Handler, vertx: Vertx, bulkOperationsEnabled: Boolean) : Handler(vertx, bulkOperationsEnabled) {
 
     override fun onPacket(buffer: ByteBuf, packet: Packet) {
-        val offset = buffer.readerIndex()
-
-        val type = buffer.getByte(offset).toInt()
-        val code = buffer.getByte(offset + 1).toInt()
+        // Type
+        val type = buffer.readByte().toInt()
```

Comment:
Is it really necessary to convert `type` and `code` to Integer?

Reply:
Yes, because we use them in `if` statement and operator `==` can't be applied to Byte